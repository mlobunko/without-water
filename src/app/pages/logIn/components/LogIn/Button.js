import React from 'react';
import { Button as ButtonStrap } from 'reactstrap';
import PropTypes from 'prop-types';

import { StyledButton } from './styles';

export const Button = ({ color, disabled, outline, text, type }) => (
  <StyledButton>
    <ButtonStrap
      outline={outline}
      color={color}
      type={type}
      disabled={disabled}
    >
      {text}
    </ButtonStrap>
  </StyledButton>
);

export default Button;

Button.propTypes = {
  color: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  outline: PropTypes.bool,
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired
};
