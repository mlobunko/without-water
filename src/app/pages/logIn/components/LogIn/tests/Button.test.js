import React from 'react';
import { shallow } from 'enzyme';

import { Button } from '../Button';

describe('render', () => {
  const props = {
    text: 'Log In',
    outline: true,
    color: 'dark',
    type: 'submit',
    disabled: false
  };
  it('renders correctly', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
