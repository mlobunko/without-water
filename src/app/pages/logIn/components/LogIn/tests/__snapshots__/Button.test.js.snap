// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledButton>
  <Button
    color="dark"
    disabled={false}
    outline={true}
    tag="button"
    type="submit"
  >
    Log In
  </Button>
</StyledButton>
`;
